src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -488,10 +494,22 @@ public void setTeamDomain(String teamDomain) {
             this.teamDomain = teamDomain;
         }
 
+        /**
+         * Deprecated for removal in 3.0
+         *
+         * Use tokenCredentialId instead
+         */
+        @Deprecated
         public String getToken() {
             return token;
```

Comment:
The getter should also retreieve the new token, so any one using the public API gets the new migrated value.

Reply:
Cool should be safe enough for now, the only caller I know of is this https://github.com/jenkinsci/global-slack-notifier-plugin/blob/master/src/main/java/org/jenkinsci/plugins/globalslack/GlobalSlackNotifier.java
and it will handle it fine.

I'll try update this next week